app/src/main/java/com/pizzapricing/Pizza.java

```diff
@@ -5,19 +5,27 @@
  */
 final class Pizza {
     private final double pizzaDiameter;
+    private final double pizzaPrice;
 
     /** Creates a new pizza with the specified diameter. */
-    Pizza(double pizzaDiameter) {
+    Pizza(double pizzaDiameter, double pizzaPrice) {
         this.pizzaDiameter = pizzaDiameter;
+        this.pizzaPrice = pizzaPrice;
     }
 
     /** Returns the diameter of the pizza. */
     double getDiameter() {
         return pizzaDiameter;
     }
 
+    /** Returns the price of the pizza. */
+    double getPrice() {return pizzaPrice; }
+
     /** Returns the total surface area of the pizza. */
     double getArea() {
         return Math.pow(pizzaDiameter/2,2)*3.1416;
     }
+
+    /** Returns price per square inch of pizza. */
+    double getCost() { return pizzaPrice/Math.pow(pizzaDiameter/2,2)*3.1416; }
```

Comment:
Can you avoid repeating the area computation that's already defined in getArea()?